SCD - Concurrently mutate operations that have different altitudes returns http 500

**Concurrent Mutate**


If we mutate tens/hundreds of operations with each has he same 2d area, but different ALTITUDE ranges CONCURRENTLY to DSS, a portion of them will fail with http 500:

```
>       assert resp.status_code == 200, resp.content
E       AssertionError: b'{
E           "error": "Internal server error E:1027f1fa-82b4-4545-92aa-ea69336f8442",
E           "code": 13,
E           "message": "Internal server error E:1027f1fa-82b4-4545-92aa-ea69336f8442",
E           "error_id": "E:6f3a07af-deff-40f5-aef5-4891746e9c72"
E         }'
E       assert 500 == 200
E        +  where 500 = <Response [500]>.status_code
```

with below error trace.

l2020-08-26T22:59:28.974Z	error	errors/errors.go:145	Uncoded error E:1027f1fa-82b4-4545-92aa-ea69336f8442 during unary server call	{"method": "/scdpb.UTMAPIUSSDSSAndUSSUSSService/PutOperationReference", "stacktrace": "Unable to SearchOperations\n --- at /app/pkg/scd/operations_handler.go:413 (Server.PutOperationReference.func1) ---\nCaused by: Error fetching Operations\n --- at /app/pkg/scd/store/cockroach/operations.go:296 (repo.searchOperations) ---\nCaused by: Error in query: \n\t\t\tSELECT\n\t\t\t\tscd_operations.id,scd_operations.owner,scd_operations.version,scd_operations.url,scd_operations.altitude_lower,scd_operations.altitude_upper,scd_operations.starts_at,scd_operations.ends_at,scd_operations.subscription_id,scd_operations.updated_at\n\t\t\tFROM\n\t\t\t\tscd_operations\n\t\t\tJOIN\n\t\t\t\t(SELECT DISTINCT\n\t\t\t\t\tscd_cells_operations.operation_id\n\t\t\t\tFROM\n\t\t\t\t\tscd_cells_operations\n\t\t\t\tWHERE\n\t\t\t\t\tscd_cells_operations.cell_id = ANY($1)\n\t\t\t\t)\n\t\t\tAS\n\t\t\t\tunique_operations\n\t\t\tON\n\t\t\t\tscd_operations.id = unique_operations.operation_id\n\t\t\tWHERE\n\t\t\t\tCOALESCE(scd_operations.altitude_upper >= $2, true)\n\t\t\tAND\n\t\t\t\tCOALESCE(scd_operations.altitude_lower <= $3, true)\n\t\t\tAND\n\t\t\t\tCOALESCE(scd_operations.ends_at >= $4, true)\n\t\t\tAND\n\t\t\t\tCOALESCE(scd_operations.starts_at <= $5, true)\n --- at /app/pkg/scd/store/cockroach/operations.go:53 (repo.fetchOperations) ---\nCaused by: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): \"sql txn\" meta={id=d0f8d22a key=/Table/62/1/\"U\\xd4ܖ\\x93\\x1dK\\a\\xa2a\\x00\\x8a\\xb9`\\xae\\x97\"/0 pri=0.01611048 epo=0 ts=1598482768.927432562,0 min=1598482768.927432562,0 seq=4} lock=true stat=ABORTED rts=1598482768.927432562,0 wto=false max=1598482768.927432562,0", "error": "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): \"sql txn\" meta={id=d0f8d22a key=/Table/62/1/\"U\\xd4ܖ\\x93\\x1dK\\a\\xa2a\\x00\\x8a\\xb9`\\xae\\x97\"/0 pri=0.01611048 epo=0 ts=1598482768.927432562,0 min=1598482768.927432562,0 seq=4} lock=true stat=ABORTED rts=1598482768.927432562,0 wto=false max=1598482768.927432562,0"}

--- a/go.mod
+++ b/go.mod
@@ -3,6 +3,8 @@ module github.com/interuss/dss
 // This forked version of openapi2proto has limited support for Open API v3.
 replace github.com/NYTimes/openapi2proto => github.com/davidsansome/openapi2proto v0.2.3-0.20190826092301-b98d13b38dab
 
+replace github.com/palantir/stacktrace => github.com/interuss/stacktrace v0.0.0-20200827180054-b2e58cf48818
+
 go 1.14
 
 require (
@@ -19,7 +21,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-middleware v1.2.0
 	github.com/grpc-ecosystem/grpc-gateway v1.14.3
 	github.com/lib/pq v1.5.2
-	github.com/palantir/stacktrace v0.0.0-20161112013806-78658fd2d177
+	github.com/palantir/stacktrace v0.0.0-00010101000000-000000000000
 	github.com/stretchr/testify v1.5.1
 	go.uber.org/multierr v1.5.0
 	go.uber.org/zap v1.15.0

--- a/pkg/errors/errors_test.go
+++ b/pkg/errors/errors_test.go
@@ -1,0 +1,14 @@
+package errors
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/palantir/stacktrace"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStacktraceUnwrap(t *testing.T) {
+	cause := errors.New("test")
+	assert.Equal(t, cause, errors.Unwrap(stacktrace.Propagate(cause, "test")))
+}

